File: packages/ts-sdk/test/adapters/mirrors.test.ts
import { describe, expect, it } from 'vitest';
import { MirrorsAdapter } from '../../src/treedx/index.js';
import { MockTransport } from './mock.js';

describe('MirrorsAdapter', () => {
  it('constructs mirror endpoints', async () => {
    const transport = new MockTransport();
    const adapter = new MirrorsAdapter({ transport });
    await adapter.list('repo');
    await adapter.upsert('repo', {});
    await adapter.sync('repo', 'mirror');
    await adapter.health('repo', 'mirror');
    await adapter.promote('repo', 'mirror');
    expect(transport.requests.map((r) => `${r.method} ${r.path}`)).toEqual([
      'GET /api/v1/repos/repo/mirrors',
      'POST /api/v1/repos/repo/mirrors',
      'POST /api/v1/repos/repo/mirrors/mirror/sync',
      'POST /api/v1/repos/repo/mirrors/mirror/health',
      'POST /api/v1/repos/repo/mirrors/mirror/promote'
    ]);
  });
});

File: packages/elixir-sdk/test/adapters/federation/mirrors_test.exs
defmodule TreeDxSdk.MirrorsAdapterTest do
  use ExUnit.Case, async: true

  test "all constructs expected request" do
    {:ok, pid} = TreeDxSdk.Test.MockTransport.start_link()
    client = TreeDxSdk.Test.MockTransport.client(pid)
    TreeDxSdk.Mirrors.list(client, "repo/a")
    TreeDxSdk.Mirrors.upsert(client, "repo/a", %{})
    TreeDxSdk.Mirrors.sync(client, "repo/a", "mir/a")
    TreeDxSdk.Mirrors.health(client, "repo/a", "mir/a")
    TreeDxSdk.Mirrors.promote(client, "repo/a", "mir/a")

    assert Enum.any?(
             TreeDxSdk.Test.MockTransport.requests(pid),
             &(&1.method == :post and &1.path == "/api/v1/repos/repo%2Fa/mirrors/mir%2Fa/promote")
           )
  end
end
